docs/documentation/automation/question_exec.md

```diff
@@ -6,4 +6,105 @@ sidebar_label: Question Exec
 import useBaseUrl from '@docusaurus/useBaseUrl'; 
 import Highlight from '@theme/Highlight';
 
-# Subtitle {#subtitle}
+<span class="hero__subtitle">
+
+_Customize Survey Questions with Javascript Code_
+
+</span>
+
+## Overview
+Surveys are among Cotalker's main features. Their flexibility and customization ability allow them to adapt to each business's needs.
+
+This spec expands the notion of survey functionality by introducing _Question EXEC_, custom code execution for your survey questions.
+
+:::info
+- Custom code must be written in **Javascript**.
+- Code should always be previously _sandboxed_.
+:::
+
+## Set Up
+
+To add your custom code, choose any survey component and select the **Automation** tab. The code is added in the field at the bottom of the settings panel.
+_See the image below:_
+
+<img alt="automation tab" class="img_sizing_narrow item shadow--md" src={useBaseUrl('img/automation_question_exec_00.png')} />
+<br/>
+<br/>
+
+You can add custom code in any of the four _stages_ of a _survey component's lifecycle_. Set the **stage** either with the dropdown menu or by clicking the corresponding icon. 
+_See the image below:_
+
+<img alt="automation tab" class="img_sizing_narrow item shadow--md" src={useBaseUrl('img/automation_question_exec_01.png')} />
+<br/>
+
+## Component Lifecycle Stages
+- **Preload**: The code executes when a new survey instance is summoned.
+- **onDisplay**: Code executes when a survey is in editing mode.
+- **Validate**: Code executes before internal validations, which permits adding additional levels of validation (business guidelines, protocols, etc.).
+- **Postsave**: Code executes after custom and internal validations.
+
+## Coding Guide
+### Available Commands:
+_The following commands are available for use in your code:_
+- **SET_READONLY**: Sets the field to read-only mode.
+- **SET_RESPONSES**: Replaces the value stored in the field.
+- **SET_REQUIRED**: Sets the field as mandatory, requiring an answer.
+
+### Parameters (Context Variables)
+_The following context variables are available for use in the code but must be selected from the Parameters menu for them to be accessible:_
+
+- **property#channel**: Properties associated with a channel.
+- **property#user**: Properties associated with a user.
+- **channel#self**: Channel where the survey is located.
+- **user#me**: User that is answering the survey.
+- **responses#self**: Survey field answer stored as value.
+- **messages#self**: Message associated with the survey.
+
```

Comment:
Add how to use these context variables
e.g.,
```
const channelId = context['channel#self']._id;   //  e.g. 606b0d554c479b00080f0d27
const channelCode = context['channel#self'].nameCode;    //  project_s_np
const channelDisplay = context['channel#self'].nameDisplay;  //  Project Secret Number Parser
```
The full model for channels is available at Channels https://doc.cotalker.com/docs/documentation/api/communication/channels

NOTE: This page (the link) has wrong names for code and display, 
it shoud be 
nameCode instead of code
and
nameDisplay instead of display

Reply:
Properties and User context model is also shown in the API documentation

Reply:
The context variables that are mentioned in this draft have a corresponding option in the *Parameters* menu. 
These other context variables now mentioned aren't displayed in the menu. 
This isn't a problem, right?